### PR TITLE
Add ResponseProperties object and example refactor

### PR DESCRIPTION
**Pull request by azgabur**
Based on commit b042b30f63e8c1040de7138bc13469217fa0493d in #212 

Usage of dynamic response function is quite common in tests to expose and check Authorino internal variables. I propose creating new object for easier definition of this function without the need to construct complicated dictionaries in tests.

**Contains fix of Value class to dataclass.**

This PR also contains example refactor of https://github.com/Kuadrant/testsuite/tree/main/testsuite/tests/kuadrant/authorino/response tests.

### CODE REVIEW

**Comment by azgabur** [comment]
After #214 will get merged I will refactor it to use ResponseProperties object.

**Comment by averevki** [comment]
Some elements from this PR were used in the reformat of the AuthConfig responses section https://github.com/Kuadrant/testsuite/pull/220.

File: testsuite/objects/__init__.py [hunk 1]
```diff
@@ -63,6 +63,53 @@ def to_dict(self):
...
+    name: str
+    properties: List[JsonProperty]
+
+    wrapper: Optional[Literal["httpHeader", "envoyDynamicMetadata"]] = None
+    wrapperKey: Optional[str] = None  # pylint: disable=invalid-name
+    wristband: Optional[dict] = None  # new class for wristband?
+
+    def to_dict(self):
+        """Returns dict representation of itself (shallow copy only)"""
+        ret = {"name": self.name, "json": {"properties": [prop.to_dict() for prop in self.properties]}}
```
[the remark this answers is not in the document]
Comment (azgabur) [reply testsuite/objects/__init__.py]: addressed

File: testsuite/objects/__init__.py [hunk 2]
```diff
@@ -35,20 +35,20 @@ class Rule:
...
+@dataclass
 class Value:
-    """Dataclass for specifying a Value in Authorization, can be either constant or value from AuthJson (jsonPath)"""
+    """Dataclass for specifying a Value in Authorization, can be either constant or value from AuthJSON (jsonPath)"""
 
-    # pylint: disable=invalid-name
-    def __init__(self, value=None, jsonPath=None) -> None:
-        super().__init__()
-        if not (value is None) ^ (jsonPath is None):
+    value: Optional[Union[str, list, dict]] = None
```
Comment (azgabur) [inline testsuite/objects/__init__.py:42]: Should the type be `Optional[Any]` ?

File: testsuite/objects/__init__.py [hunk 3]
```diff
@@ -63,6 +63,92 @@ def to_dict(self):
...
+    https://github.com/kuadrant/authorino/blob/v0.13.0/api/v1beta1/auth_config_types.go#L89
+    Dataclass which is a named `Value` class
+    For use in identity extended_property functionality and Response functionality.
+    """
+
+    name: str = "property-name"
+
+    def to_dict(self):
+        """Returns dict representation of itself (shallow copy only)"""
+        return {"name": self.name, **super().to_dict()}
```
Comment (jsmolar) [inline testsuite/objects/__init__.py:78]: '**super().to_dict()' could be done better

File: testsuite/objects/__init__.py [hunk 4]
```diff
@@ -63,6 +63,92 @@ def to_dict(self):
...
+
+
+@dataclass
+class ExtendedProperty(JsonProperty):
```
Comment (jsmolar) [inline testsuite/objects/__init__.py:82]: `ExtendedProperty` has only one additional "feature" then `JsonProperty` (`overwrite`)?

Comment (azgabur) [reply testsuite/objects/__init__.py:82]: Yes, I am mimicking source code classes https://github.com/Kuadrant/authorino/blob/v0.14.0/api/v1beta1/auth_config_types.go#L190

**Comment by azgabur** [comment]
Closing in favor of #220

**Pull request closed without merging**